setup.py

```diff
@@ -48,7 +56,9 @@
                                   '_static/metpy_150x150.png', '_static/unidata_75x75.png',
                                   '_static/unidata_150x150.png'],
                   'metpy': ['static-data-manifest.txt']},
-
+    use_scm_version={'write_to': 'metpy/_version.py',
+                     'write_to_template': "version = '{version}'"},
+    setup_requires=['setuptools_scm'],
```

Comment:
Having this here in `setup_requires` means that `easy_install` is used to download and install, which is what puts it in the `.eggs` directory and breaks some of our checking, not to mention is super annoying.

Reply:
Your overall review comment may address this if we can get things moved to `setup.cfg`. Didn't know the advantages of why one should do that.

Reply:
Just trying to reduce the complexity and the amount of arbitrary python code being run in `setup.py` in favor of declaring metadata and options in `setup.cfg`. IMO, it makes things more maintainable.